packages/flutter_tools/bin/podhelper.rb

```diff
@@ -65,12 +65,34 @@ def flutter_additional_ios_build_settings(target)
   # Bundles are com.apple.product-type.bundle, frameworks are com.apple.product-type.framework.
   target_is_resource_bundle = target.respond_to?(:product_type) && target.product_type == 'com.apple.product-type.bundle'
 
+  # Detect pod-level EXCLUDED_ARCHS for the simulator written to Target Support Files xcconfig.
+  support_files_dir = File.join('Pods', 'Target Support Files', target.name)
+  if Dir.exist?(support_files_dir)
+    Dir.glob(File.join(support_files_dir, '*.xcconfig')).each do |xcfile|
+      File.foreach(xcfile) do |line|
+        if line =~ /^\s*EXCLUDED_ARCHS\b/ && line.include?('arm64')
```

Comment:
So I think we should limit this to only Xcode 26. Looks like you should be able to get the Xcode version from the `XCODE_VERSION_MAJOR` build setting.